src/include/sof/lib/alloc.h

```diff
@@ -59,6 +59,7 @@ enum mem_zone {
 	SOF_MEM_ZONE_BUFFER,		/**< Buffer zone */
 	SOF_MEM_ZONE_RUNTIME_SHARED,	/**< Runtime shared zone */
 	SOF_MEM_ZONE_SYS_SHARED,	/**< System shared zone */
+	SOF_MEM_ZONE_SIZE           /**< Size of the mem zone capabilities table */
```

Comment:
Do you think we could use the Zephyr allocator natively in the first step and wrap the xtos versions (this could simplify the new allocator APIs as we could wrap them to fall back to existing APIs in the xtos allocator - meaning there would be no restriction in usage of new APIs)